### PR TITLE
Refactor Views

### PR DESCRIPTION
**Pull request by GoodBoyDigital**
As v8 has evolved. It has become clear that seperating out view and container logic has caused unnecessary complications to the code. 

This refactor does away with the `View` classes and instead All renderalbe things now just extend Container. This makes things much simpler to follow.

### CODE REVIEW

File: src/rendering/renderers/shared/system/SharedSystems.ts [hunk 1]
```diff
@@ -25,14 +28,17 @@ export const SharedSystems = [
     TextureGCSystem,
     GenerateTextureSystem,
     ExtractSystem,
+    HTMLTextSystem
```
Comment (Zyie) [inline src/rendering/renderers/shared/system/SharedSystems.ts:31]: This gets added in the `init.ts` for html text, so shouldn't be added here

File: src/rendering/renderers/shared/system/SharedSystems.ts [hunk 2]
```diff
@@ -25,14 +28,17 @@ export const SharedSystems = [
...
 export const SharedRenderPipes = [
     BlendModePipe,
     BatcherPipe,
     SpritePipe,
+    NineSliceSpritePipe,
     RenderGroupPipe,
     AlphaMaskPipe,
     StencilMaskPipe,
     ColorMaskPipe,
+    TilingSpritePipe,
```
Comment (Zyie) [inline src/rendering/renderers/shared/system/SharedSystems.ts:43]: This gets added in the init.ts for tiling sprites, so shouldn't be added here

File: src/scene/container/RenderGroupSystem.ts [hunk 3]
```diff
@@ -40,6 +40,12 @@ export class RenderGroupSystem implements System
     {
         container.isRenderGroup = true;
 
+        const parent = container.parent;
+
+        container.parent = null;
+
+        //  container.__rooty = true;
```
Comment (Zyie) [inline src/scene/container/RenderGroupSystem.ts:47]:
```suggestion
```

File: src/scene/container/RenderGroupSystem.ts [hunk 4]
```diff
@@ -40,6 +40,11 @@ export class RenderGroupSystem implements System
     {
         container.isRenderGroup = true;
 
+        const parent = container.parent;
+
+        container.parent = null;
+
+
```
Comment (Zyie) [inline src/scene/container/RenderGroupSystem.ts:47]:
```suggestion
```

File: src/rendering/renderers/shared/system/SharedSystems.ts (hunk 1 above)
Comment (GoodBoyDigital) [reply src/rendering/renderers/shared/system/SharedSystems.ts:31]: cool, removed!

File: src/rendering/renderers/shared/system/SharedSystems.ts (hunk 2 above)
Comment (GoodBoyDigital) [reply src/rendering/renderers/shared/system/SharedSystems.ts:43]: ta, removed!

File: src/scene/graphics/shared/Graphics.ts [hunk 5]
```diff
@@ -22,14 +26,10 @@ import type { ConvertedFillStyle, ConvertedStrokeStyle, FillStyle, FillStyleInpu
...
  */
-export interface GraphicsOptions extends Partial<ContainerOptions<GraphicsView>>
+export interface GraphicsOptions extends ContainerOptions
 {
     /** The GraphicsContext to use, useful for reuse and optimisation */
     context?: GraphicsContext;
-    /** The fill style to use */
-    fillStyle?: FillStyleInputs;
-    /** The stroke style to use */
-    strokeStyle?: FillStyleInputs;
```
Comment (Zyie) [inline src/scene/graphics/shared/Graphics.ts:32]: why have these been removed?

we have getter/setters for these on the Graphics class os they should be part of the constructor options?

File: src/scene/sprite-nine-slice/NiceSliceSpritePipe.ts [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    private readonly _gpuSpriteHash: Record<number, BatchableMesh> = Object.create(null);
+
+    constructor(renderer: Renderer)
+    {
+        this._renderer = renderer;
+    }
+
+    public addRenderable(sprite: NineSliceSprite, _instructionSet: InstructionSet)
+    {
+        const gpuSprite = this._getGpuSprite(sprite);
```
Comment (Zyie) [inline src/scene/sprite-nine-slice/NiceSliceSpritePipe.ts]: some of these functions have exactly the same code as `SpritePipe`, should we extract these out into reusable functions?

Comment (Zyie) [reply src/scene/sprite-nine-slice/NiceSliceSpritePipe.ts]: to be followed up in a separate pr

**Review by Zyie — APPROVED** [review]

**Pull request merged**